src/TwitterRoomHandler.js

```diff
@@ -0,0 +1,109 @@
+/*
+  Hander for the many different types of rooms this AS will support.
+  Types are as follows (which are tracked in 'twitter_type' in the remote rooms):
+  'timeline' -  A room that tracks a singular users activities
+  'service' -   A room that can link/unlink twitter accounts from users and
+                provide general assistance.
+
+*/
+var RemoteRoom = require("matrix-appservice-bridge").RemoteRoom;
+var log = require('npmlog');
+var TwitterHandler = require('./TwitterHandler.js').TwitterHandler;
+
+
+/**
+ * TwitterRoomHandler - The handler class to delegate which handler should Deal
+ * with which request.
+ * @class
+ * @param  {matrix-appservice-bridge.Bridge} bridge
+ * @param  {Object.<string,TwitterHandler>} handlers Handers to register for
+ * each type of room.
+ */
+var TwitterRoomHandler = function (bridge, handlers) {
+  this._bridge = bridge;
+  this.handlers = handlers; // 'service' handler
+}
+
+TwitterRoomHandler.prototype.processInvite = function (event,request, context){
+  var remote = context.rooms.remote;
+  if(remote == null
+     && event.sender != "@twitbot:localhost"
+     && event.state_key == "@twitbot:localhost")
```

Comment:
`localhost` references should be nuked. We want people to run this on actual HSes remember?